### PR TITLE
Add projected gradient convenience class.

### PR DESCRIPTION
**Pull request by mblondel**

### CODE REVIEW

File: jaxopt/_src/projected_gradient.py [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+@dataclass
+class ProjectedGradient(base.IterativeSolverMixin):
+  """Projected gradient solver.
+
+  This solver is a convenience wrapper around ``ProximalGradient``.
+
+  Attributes:
+    fun: a smooth function of the form ``fun(parameters, *args, **kwargs)``,
+      where ``parameters`` are the model parameters w.r.t. which we minimize
+      the function andethe rest are fixed auxiliary parameters.
```
Comment (fllinares) [inline jaxopt/_src/projected_gradient.py:39]: and the

File: jaxopt/_src/projected_gradient.py [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+    """
+    return self._pg.init(init_params, hyperparams_proj, *args, **kwargs)
+
+  def update(self,
+             params: Any,
+             state: NamedTuple,
+             hyperparams_proj: Optional[Any] = None,
+             *args,
+             **kwargs) -> base.OptStep:
+    """Performs one iteration of proximal gradient.
```
Comment (fllinares) [inline jaxopt/_src/projected_gradient.py:94]: typo: projected

File: tests/projected_gradient_test.py [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+    rng = onp.random.RandomState(0)
+    X = rng.randn(10, 5)
+    w = rng.rand(5)
+    y = jnp.dot(X, w)
+    fun = objective.least_squares
+    w_init = jnp.zeros_like(w)
+
+    pg = ProjectedGradient(fun=fun,
+                           projection=projection.projection_non_negative)
+    pg_sol = pg.run(w_init, data=(X,y)).params
```
Comment (fllinares) [inline tests/projected_gradient_test.py:43]: nitpick: space after comma for consistency

File: tests/projected_gradient_test.py [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+
+    lbfgsb = ScipyBoundedMinimize(fun=fun, method="l-bfgs-b")
+    lower_bounds = jnp.zeros_like(w_init)
+    upper_bounds = jnp.ones_like(w_init) * jnp.inf
+    bounds = (lower_bounds, upper_bounds)
+    lbfgsb_sol = lbfgsb.run(w_init, bounds=bounds, data=(X,y)).params
```
Comment (fllinares) [inline tests/projected_gradient_test.py:49]: same

**Review by fllinares — APPROVED** [review]
Thanks a lot Mathieu, everything LGTM. Just spotted a couple of typos.

One question though: what's the main reason for `ProjectedGradient` not to inherit from `ProximalGradient`? Mainly asking for myself to be consistent with the design when implementing other solvers in the future.
